File: 2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.rb
# @param {String} word1
# @param {String} word2
# @return {Boolean}
def check_almost_equivalent(word1, word2)
  char_counts = Hash.new(0)
  word1.each_char { |c| char_counts[c] += 1 }
  word2.each_char { |c| char_counts[c] -= 1 }
  char_counts.values.all? { |v| v.abs <= 3 }
end

File: 2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.py
class Solution:
    def checkAlmostEquivalent(self, word1: str, word2: str) -> bool:
        charCounts = defaultdict(int)
        for c in word1: charCounts[c]+=1
        for c in word2: charCounts[c]-=1
        return all( abs(val) <= 3 for val in charCounts.values() )



       



